### PR TITLE
Mahmoud new

### PR DESCRIPTION
**Pull request by MahmoudM14**
Pull request content the following: 
1- split the script to two scripts autoBugTracker.py (main script) and executeUserScript.py to be responsible of execution only. 
2- distinguish between missing lib, script not available or problematic script 
3- in case of script failure (where we need to submit bug), I've return a list of the traceback to be included in the email going to the triage team. 

There is ends to be joined after the database implemented. 
Please let me know if you have any questions.
------------
[AB#85](https://dev.azure.com/dhk2/65e21a49-896d-470d-8659-fc7cca2f4625/_workitems/edit/85) @ismustachio @cam28 @rag4 
[AB#68](https://dev.azure.com/dhk2/65e21a49-896d-470d-8659-fc7cca2f4625/_workitems/edit/68) @ismustachio @TonyDiMaggio @prikim 
[AB#66](https://dev.azure.com/dhk2/65e21a49-896d-470d-8659-fc7cca2f4625/_workitems/edit/66) @prikim @rag4 @TonyDiMaggio

### CODE REVIEW

File: autoBugTracker.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import argparse
+import executeUserScript
+
+
+class autoBugTracker(object):
```
Comment (DanaKhoshnaw) [inline autoBugTracker.py:5]: I believe we agreed on a more conventional name such as Main

File: autoBugTracker.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    def __init__(self):
+        print("***Auto Bug Tracker***\n")
+        self.execute = executeUserScript.executeUserScript()
+
+
+    def parsingCommandLineArguments(self):
+        '''' Return dictionary
```
Comment (DanaKhoshnaw) [inline autoBugTracker.py:12]: Please follow the same commenting style like the rest of the project

File: executeUserScript.py [hunk 3]
```diff
@@ -1,47 +1,22 @@
...
-            :return
-                dictionary
-        '''
-        parser = argparse.ArgumentParser()
-        parser.add_argument('-S', '--userscript', help='User script Required!', required=True)
-        arguments = parser.parse_args()
-        return vars(arguments)
+class executeUserScript(object):
 
     def executeScript(self, scriptName):
```
Comment (DanaKhoshnaw) [inline executeUserScript.py:6]: Are we only allowing a script to be used to run our program? Can they not provide command-line arguments to call and start their program?

File: autoBugTracker.py (hunk 1 above)
Comment (MahmoudM14) [reply autoBugTracker.py:5]: That was stretch goal when creating a package. I would discuss it again with product owner.

File: autoBugTracker.py (hunk 2 above)
Comment (MahmoudM14) [reply autoBugTracker.py:12]: By Product owner follow the comments style of Python 3.8. Document.

File: executeUserScript.py (hunk 3 above)
Comment (MahmoudM14) [reply executeUserScript.py:6]: is this discussed already? do you want to bring it up to the next meeting? Up to now this was the agreement. This should not stop this pull request.

File: autoBugTracker.py (hunk 2 above)
Comment (DanaKhoshnaw) [reply autoBugTracker.py:12]: I thought 3.8 was with double quotes instead of single quotes?

File: executeUserScript.py (hunk 3 above)
Comment (DanaKhoshnaw) [reply executeUserScript.py:6]: Not blocking, will need to get clarification. But could you please rename the file to match the rest of the project?

Comment (MahmoudM14) [reply executeUserScript.py]: Naming of the files have been changed. Please let me know if you have any question. 

Thanks!

**Pull request merged**